### PR TITLE
common: improve unit test coverage of 'kv' package

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Improve unit test coverage of 'kv' package

### CODE REVIEW

File: lightning/kv/sql2kv_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	minNotNull := types.Datum{}
+	minNotNull.SetMinNotNull()
+	datums := rowArrayMarshaler{types.NewStringDatum("1"), nullDatum, minNotNull, types.MaxValueDatum()}
+	encoder := &mockEncoder{buf: &bytes.Buffer{}}
+	err := datums.MarshalLogArray(encoder)
+	c.Assert(err, IsNil)
+	c.Assert(encoder.buf.String(), Equals, "[col=0][kind=string][val=1][col=1][kind=null][val=NULL][col=2][kind=min][val=-inf][col=3][kind=max][val=+inf]")
+
+	invalid := []types.Datum{nullDatum, minNotNull}
+	pkind := (*byte)(unsafe.Pointer(&invalid[1]))
```
Comment (kennytm) [inline lightning/kv/sql2kv_test.go]: Please use `SetInterface` to create a datum which `ToString()` would fail.

Comment (lonng) [reply lightning/kv/sql2kv_test.go]: addressed

File: lightning/kv/sql2kv_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	buf *bytes.Buffer
+}
+
+func (enc *mockEncoder) AppendObject(obj zapcore.ObjectMarshaler) error {
+	err := obj.MarshalLogObject(enc)
+	return err
+}
+
+func (enc *mockEncoder) AddInt(key string, value int) {
+	enc.buf.WriteString(fmt.Sprintf("[%v=%v]", key, value))
```
Comment (kennytm) [inline lightning/kv/sql2kv_test.go]:
```suggestion
	fmt.Fprintf(enc.buf, "[%v=%v]", key, value)
```

File: lightning/kv/sql2kv_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func (enc *mockEncoder) AddString(key string, value string) {
+	enc.buf.WriteString(fmt.Sprintf("[%v=%v]", key, value))
```
Comment (kennytm) [inline lightning/kv/sql2kv_test.go]:
```suggestion
	fmt.Fprintf(enc.buf, "[%v=%v]", key, value)
```

File: lightning/kv/sql2kv_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func (s *kvSuite) TestMarshal(c *C) {
+	nullDatum := types.Datum{}
+	nullDatum.SetNull()
+	minNotNull := types.Datum{}
+	minNotNull.SetMinNotNull()
+	datums := rowArrayMarshaler{types.NewStringDatum("1"), nullDatum, minNotNull, types.MaxValueDatum()}
+	encoder := &mockEncoder{buf: &bytes.Buffer{}}
```
Comment (kennytm) [inline lightning/kv/sql2kv_test.go]: I suggest we just use a [`MapObjectEncoder`](https://godoc.org/go.uber.org/zap/zapcore#MapObjectEncoder) here so you don't even need `mockEncoder`.

```go
	encoder := zapcore.NewMapObjectEncoder()
	encoder.AddArray("test", &S{})
	c.Assert(encoder.Fields["test"], DeepEquals, []map[string]interface{} { ... })
```

**Comment by lonng** [comment]
@kennytm @july2993 PTAL

File: lightning/kv/sql2kv_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	logger := log.Logger{Logger: zap.NewNop()}
+	rows := []types.Datum{
+		types.NewIntDatum(10000000),
+	}
+
+	// Strict mode
+	strictMode := NewTableKVEncoder(tbl, mysql.ModeStrictAllTables)
+	pairs, err := strictMode.Encode(logger, rows, 1, []int{0, 1})
+	c.Assert(err, NotNil)
+	c.Assert(err, ErrorMatches, ".*overflows tinyint")
```
Comment (kennytm) [inline lightning/kv/sql2kv_test.go]: If you use ErrorMatches you don't need NotNil.

File: lightning/kv/session_test.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+func TestKV(t *testing.T) {
+	TestingT(t)
+}
+
+func (s *kvSuite) TestSetOption(c *C) {
+	session := newSession(mysql.ModeNone)
+	txn, err := session.Txn(true)
+	c.Assert(err, IsNil)
+	txn.SetOption(tidbkv.Priority, tidbkv.PriorityHigh)
```
[the remark this answers is not in the document]
Comment (lonng) [reply lightning/kv/session_test.go]: Yes, just

**Pull request merged**